https://github.com/OpenSMTPD/OpenSMTPD/pull/1198

From 84331a266b7d8d8e469aea8b85d1e493725807ae Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Thu, 13 Apr 2023 11:19:53 -0700
Subject: [PATCH 1/2] build: Fix -Werror=strict-prototypes

Clang-16 is more strict about these so make the compiler happy.
---
 usr.sbin/smtpd/bounce.c | 2 +-
 usr.sbin/smtpd/ioev.c   | 2 +-
 usr.sbin/smtpd/mda.c    | 4 ++--
 usr.sbin/smtpd/smtpd.c  | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/usr.sbin/smtpd/bounce.c b/usr.sbin/smtpd/bounce.c
index bb08f90e..7fb1cf32 100644
--- a/usr.sbin/smtpd/bounce.c
+++ b/usr.sbin/smtpd/bounce.c
@@ -243,7 +243,7 @@ bounce_timeout(int fd, short ev, void *arg)
 }
 
 static void
-bounce_drain()
+bounce_drain(void)
 {
 	struct bounce_message	*msg;
 	struct timeval		 tv;
diff --git a/usr.sbin/smtpd/ioev.c b/usr.sbin/smtpd/ioev.c
index 747b3a51..3e3e68f4 100644
--- a/usr.sbin/smtpd/ioev.c
+++ b/usr.sbin/smtpd/ioev.c
@@ -228,7 +228,7 @@ io_frame_leave(struct io *io)
 }
 
 void
-_io_init()
+_io_init(void)
 {
 	static int init = 0;
 
diff --git a/usr.sbin/smtpd/mda.c b/usr.sbin/smtpd/mda.c
index 6a28b923..bd963e54 100644
--- a/usr.sbin/smtpd/mda.c
+++ b/usr.sbin/smtpd/mda.c
@@ -386,12 +386,12 @@ mda_imsg(struct mproc *p, struct imsg *imsg)
 }
 
 void
-mda_postfork()
+mda_postfork(void)
 {
 }
 
 void
-mda_postprivdrop()
+mda_postprivdrop(void)
 {
 	tree_init(&sessions);
 	tree_init(&users);
diff --git a/usr.sbin/smtpd/smtpd.c b/usr.sbin/smtpd/smtpd.c
index 26078c49..830417e6 100644
--- a/usr.sbin/smtpd/smtpd.c
+++ b/usr.sbin/smtpd/smtpd.c
@@ -353,7 +353,7 @@ parent_send_config_dispatcher(void)
 }
 
 void
-parent_send_config_lka()
+parent_send_config_lka(void)
 {
 	log_debug("debug: parent_send_config_ruleset: reloading");
 	m_compose(p_lka, IMSG_CONF_START, 0, 0, -1, NULL, 0);

